custom-email-module/src/main/java/org/wso2/carbon/custom/email/HTMLEmailSendingModule.java

```diff
@@ -0,0 +1,177 @@
+            String smtpAuth =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_AUTH).getValue().toString();
+            final String smtpUser = transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_USER).getValue()
+                    .toString();
+            final String smtpPassword =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_PASSWORD).getValue().toString();
+            String smtpFrom =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_FROM).getValue().toString();
+            String contentType;
+            if (transportOutDescription.getParameter(EmailConstants.MAIL_CONTENT_TYPE) != null) {
+                contentType =
+                        transportOutDescription.getParameter(EmailConstants.MAIL_CONTENT_TYPE).getValue().toString();
+            } else {
+                contentType = "text/plain";
+            }
+
+            Properties properties = new Properties();
+            properties.put(EmailConstants.MAIL_SMTP_HOST, smtpHost);
+            properties.put(EmailConstants.MAIL_SMTP_PORT, smtpPort);
+            properties.put(EmailConstants.MAIL_SMTP_STARTTLS, smtpStarttls);
+            properties.put(EmailConstants.MAIL_SMTP_AUTH, smtpAuth);
+
+            Session session = Session.getInstance(properties,
+                    new Authenticator() {
+                        protected PasswordAuthentication getPasswordAuthentication() {
+                            return new PasswordAuthentication(smtpUser, smtpPassword);
+                        }
+                    });
+
+            Message message = new MimeMessage(session);
+            Address[] addresses = InternetAddress.parse(notification.getSendTo());
+            message.setFrom(new InternetAddress(smtpFrom));
+            message.setRecipients(Message.RecipientType.TO, addresses);
+            message.setSubject(notification.getSubject());
+            StringBuilder contents = new StringBuilder();
+            contents.append(notification.getBody())
+                    .append(System.getProperty("line.separator"))
+                    .append(System.getProperty("line.separator"))
+                    .append(notification.getFooter());
+            message.setContent(contents.toString(), contentType);
+            Thread.currentThread().setContextClassLoader(MailUtils.class.getClassLoader());
+            Transport.send(message);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Email content : " + notification.getBody());
+            }
+
+            log.info("Email notification has been sent to " + notification.getSendTo());
+        } catch (InterruptedException e) {
+            log.error("Interrupted while waiting until an element becomes available in the notification queue.", e);
+        } catch (AddressException e) {
+            log.error("Error occurred while parsing the send to email address to:", e);
```

Comment:
Try to add some context into the error logs. Insert variables that are present and can be exposed.